src/client/cli/cmd/mount.cpp

```diff
@@ -90,6 +90,22 @@ mp::ReturnCode cmd::Mount::run(mp::ArgParser* parser)
             spinner.print(cerr, reply.log_line());
         }
 
+        if (reply.credentials_requested())
+        {
+            MountRequest request;
+
+            spinner.stop();
+
+            const auto [username, password] = mcp::get_user_password(term);
+
+            request.mutable_user_credentials()->set_username(username);
+            request.mutable_user_credentials()->set_password(password);
+
+            client->Write(request);
+
+            return;
+        }
+
```

Comment:
If I am correct, this code is repeated verbatim in `restart.cpp` and `start.cpp`. Could this fragment be refactored into a new function?

Reply:
Well, it's not *quite* verbatim, the request type is different in each case.  I did give it some thought on factoring this out, but then we'd have to template the function and that gets messing with a free function, so then I'd need to include it in a class and implement it in the header.  I agree it seems quite redundant.  I'll give it a bit more thought.

Reply:
Oh, right, I didn't note that. I'm OK with repeating few lines of code in that case.

Reply:
Ok, refactored this some more.